Upgrade MinIO to support accepting non signed requests

azat opened the pull request:
During fixing one issue with multiple reads of input files while sending them to S3 this had been pops up.

The problem is in this place [1].

  [1]: https://github.com/minio/minio/blob/200caab82b622a0ec6c9776ee9b555aedbc9b320/cmd/object-handlers.go#L1573-L1591

And this had been added only in [2].

  [2]: https://github.com/minio/minio/pull/16484

But even this implementation is not complete, since it does not correctly support STREAMING-UNSIGNED-PAYLOAD-TRAILER, but this will be the first step, to see if there are still problems with newer MinIO.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

azat committed:
Upgrade MinIO to support accepting non signed requests

During fixing one issue with multiple reads of input files while sending
them to S3 this had been pops up.

The problem is in this place [1].

  [1]: https://github.com/minio/minio/blob/200caab82b622a0ec6c9776ee9b555aedbc9b320/cmd/object-handlers.go#L1573-L1591

And this had been added only in [2].

  [2]: https://github.com/minio/minio/pull/16484

But even this implementation is not complete, since it does not
correctly support STREAMING-UNSIGNED-PAYLOAD-TRAILER, but this will be
the first step, to see if there are still problems with newer MinIO.

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

robot-ch-test-poll1 commented:
<!-- automatic status comment for PR #55245 from azat/ClickHouse:bump-minio -->
*This is an automated comment for commit 918d65d6707c69ab541cdb56a076cdb83845d3ed with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55245/918d65d6707c69ab541cdb56a076cdb83845d3ed/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6410994230">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55245/918d65d6707c69ab541cdb56a076cdb83845d3ed/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

azat commented:
Interesting, ASan founds some leaks here - https://s3.amazonaws.com/clickhouse-test-reports/55245/918d65d6707c69ab541cdb56a076cdb83845d3ed/integration_tests__asan__[5_6]/integration_run_parallel3_0.log

azat commented:
>[Integration tests (asan) [5/6] — fail: 2, passed: 536](https://s3.amazonaws.com/clickhouse-test-reports/55245/918d65d6707c69ab541cdb56a076cdb83845d3ed/integration_tests__asan__[5_6].html)

- `test_storage_delta/test.py::test_multiple_log_files`

Something odd happens:

```
(Pdb) p minio_client.fput_object(bucket_name='root', object_name='/test_multiple_log_files/_delta_log/00000000000000000001.json', file_path='/test_multiple_log_files/_delta_log/00000000000000000001.json').http_headers
HTTPHeaderDict({'Accept-Ranges': 'bytes', 'Content-Length': '0', 'ETag': '"f1f3276c787b3f6724b9a8edb313c4fa"', 'Server': 'MinIO', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Vary': 'Origin, Accept-Encoding', 'X-Amz-Id-2': '7987905dee74cdeb212432486a178e511309594cee7cb75f892cd53e35f09ea4', 'X-Amz-Request-Id': '178B9B0904A9640D', 'X-Content-Type-Options': 'nosniff', 'X-Xss-Protection': '1; mode=block', 'Date': 'Fri, 06 Oct 2023 19:22:27 GMT'})
(Pdb) p list(minio_client.list_objects('root', '/test_multiple_log_files/_delta_log/'))
[]
```

azat committed:
Adjust the test_storage_delta::test_multiple_log_files after MinIO upgrade

After upgrade MinIO stopped handling of objects with leading slash:

    (Pdb) p minio_client.fput_object(bucket_name='root', object_name='/test_multiple_log_files/_delta_log/00000000000000000001.json', file_path='/test_multiple_log_files/_delta_log/00000000000000000001.json').http_headers
    HTTPHeaderDict({'Accept-Ranges': 'bytes', 'Content-Length': '0', 'ETag': '"f1f3276c787b3f6724b9a8edb313c4fa"', 'Server': 'MinIO', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Vary': 'Origin, Accept-Encoding', 'X-Amz-Id-2': '7987905dee74cdeb212432486a178e511309594cee7cb75f892cd53e35f09ea4', 'X-Amz-Request-Id': '178B9B0904A9640D', 'X-Content-Type-Options': 'nosniff', 'X-Xss-Protection': '1; mode=block', 'Date': 'Fri, 06 Oct 2023 19:22:27 GMT'})
    (Pdb) p list(minio_client.list_objects('root', '/test_multiple_log_files/_delta_log/'))
    []
    (Pdb) p minio_client.get_object(bucket_name='root', object_name='/test_multiple_log_files/_delta_log/00000000000000000001.json').read()
    b'{"commitInfo":{"timestamp":1696618985577,"operation":"WRITE","operationParameters":{"mode":"Overwrite","partitionBy":"[]"},"readVersion":0,"isolationLevel":"Serializable","isBlindAppend":false,"operationMetrics":{"numFiles":"1","numOutputRows":"100","numOutputBytes":"2459"},"engineInfo":"Apache-Spark/3.3.2 Delta-Lake/2.3.0","txnId":"2a69a4c9-f72e-4ac9-9374-bdff96707edf"}}\n{"add":{"path":"part-00000-9ff4728b-d5fe-49ac-a773-e2d6e01f4139-c000.parquet","partitionValues":{},"size":2459,"modificationTime":1696618985310,"dataChange":true,"stats":"{\\"numRecords\\":100,\\"minValues\\":{\\"col-ab7641d2-f0a9-4ffb-ae37-27087d13dfc8\\":0,\\"col-6cbfced4-5921-4e3b-9226-558171c11949\\":\\"1\\"},\\"maxValues\\":{\\"col-ab7641d2-f0a9-4ffb-ae37-27087d13dfc8\\":99,\\"col-6cbfced4-5921-4e3b-9226-558171c11949\\":\\"99\\"},\\"nullCount\\":{\\"col-ab7641d2-f0a9-4ffb-ae37-27087d13dfc8\\":0,\\"col-6cbfced4-5921-4e3b-9226-558171c11949\\":0}}"}}\n{"remove":{"path":"part-00000-7c29f5d9-e14d-419a-a273-58d07775e1e9-c000.parquet","deletionTimestamp":1696618985576,"dataChange":true,"extendedFileMetadata":true,"partitionValues":{},"size":2459}}\n'
    (Pdb) p list(minio_client.list_objects(bucket_name='root', prefix='test_multiple_log_files/', recursive=True))
    [<minio.datatypes.Object object at 0x7ffff1ec2fe0>, <minio.datatypes.Object object at 0x7ffff1ec39d0>, <minio.datatypes.Object object at 0x7ffff1ec2590>, <minio.datatypes.Object object at 0x7ffff1ec3dc0>, <minio.datatypes.Object object at 0x7ffff1ec3a00>, <minio.datatypes.Object object at 0x7ffff1ec3fa0>, <minio.datatypes.Object object at 0x7ffff1ec3f40>, <minio.datatypes.Object object at 0x7ffff1ec2770>]

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
>Integration tests (asan) [1/6] — fail: 2, passed: 393

- `test_zookeeper_config/test.py::test_chroot_with_same_root`

```
E           subprocess.TimeoutExpired: Command '['docker-compose', '--env-file', '/ClickHouse/tests/integration/test_zookeeper_config/_instances_0/.env', '--project-name', 'roottestzookeeperconfig', '--file', '/compose/docker_compose_keeper.yml', '--verbose', 'up', '-d']' timed out after 300 seconds
```
